### PR TITLE
Removed ERDAPP link and attribute, rerouted getters to new function.

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

File: src/Entity/DatasetSubmission.php [hunk 1]
```diff
@@ -2764,14 +2754,38 @@ public function getDistributionPoints()
...
+
+        return null;
+    }
+
     /**
      * Getter for the erddap url.
      *
      * @return string|null
      */
     public function getErddapUrl(): ?string
```
Comment (praneethpr) [inline src/Entity/DatasetSubmission.php:2780]: Are we using this getter anywhere?

Comment (mickel1138) [reply src/Entity/DatasetSubmission.php:2780]: No, not currently, but can be used later on the data landing page or, or whoever wants the link. I should probably make getErddapDatasetLink public, incase someone wants to use that.

File: src/Entity/DatasetSubmission.php [hunk 2]
```diff
@@ -2764,14 +2754,38 @@ public function getDistributionPoints()
...
 
+     /**
+     * Getter for the (first) erddap url link.
+     *
+     * @return DatasetLink|null
+     */
+    private function getErdappUrlLink() : ?DatasetLink
+    {
+        $datasetLinks = $this->getDatasetLinks()->filter(function (DatasetLink $datasetLink) {
+            return $datasetLink->getName() == "ERDDAP";
```
Comment (praneethpr) [inline src/Entity/DatasetSubmission.php:2765]: Can use the constant LINK_NAME_CODES from DatasetLinks Entity

File: src/Entity/DatasetSubmission.php (hunk 1 above)
Comment (praneethpr) [reply src/Entity/DatasetSubmission.php:2780]: Okay, makes sense. I think all the ERDDAP stuff should be moved to DatasetLinks entity

File: src/Entity/DatasetSubmission.php [hunk 3]
```diff
@@ -992,15 +992,6 @@ class DatasetSubmission extends Entity
...
     protected $distributionPoints;
 
-    /**
-     * ERDDAPP Url for the dataset.
-     *
-     * @var string
-     *
-     * @ORM\Column(type="text", nullable=true)
-     */
-    protected $erddapUrl;
```
Comment (praneethpr) [inline src/Entity/DatasetSubmission.php:1002]: I think there is a missing migration, and what do we want to do about the existing data in those fields?

File: src/Entity/DatasetSubmission.php (hunk 2 above)
Comment (mickel1138) [reply src/Entity/DatasetSubmission.php:2765]: Good one!

File: src/Entity/DatasetSubmission.php (hunk 3 above)
Comment (mickel1138) [reply src/Entity/DatasetSubmission.php:1002]: Yes, forgot to put that in.

File: src/Entity/DatasetSubmission.php (hunk 1 above)
Comment (mickel1138) [reply src/Entity/DatasetSubmission.php:2780]: So these are actually used in the ISO metadata generator.
Also in Dataland, but the button is not visible.

**Pull request merged**